### ISSUE TITLE
Too many ERR_MSG_TIMED_OUT generated even though the broker(s) are still UP

### ISSUE THREAD

**Issue by yhsiung** [edenhill/librdkafka#2842]
Read the FAQ first: https://github.com/edenhill/librdkafka/wiki/FAQ

Description
===========
Why so many ERR_MSG_TIMED_OUT are generated in the dr_msg_cb() even though the brokers are UP and running and connection is still established.  What else may cause the message timed out. 
The message.timeout.ms is set to 120000 (2 minutes), which should be longer enough to push the messages out.

According to the https://github.com/edenhill/librdkafka/blob/master/INTRODUCTION.md.
Internal error ERR__MSG_TIMED_OUT.
The message could not be successfully transmitted before message.timeout.ms expired, typically due to no leader being available or no broker connection. 

How to reproduce
================
<your steps how to reproduce goes here, or remove section if not relevant>

**IMPORTANT**: Always try to reproduce the issue on the latest released version (see https://github.com/edenhill/librdkafka/releases), if it can't be reproduced on the latest version the issue has been fixed.

Checklist
=========

**IMPORTANT**: We will close issues where the checklist has not been completed.

Please provide the following information:

 - [x] librdkafka version (release number or git tag): `<REPLACE with e.g., v0.10.5 or a git sha. NOT "latest" or "current">`
 - [ ] Apache Kafka version: `<REPLACE with e.g., 0.10.2.3>`
 - [ ] librdkafka client configuration: `<REPLACE with e.g., message.timeout.ms=123, auto.reset.offset=earliest, ..>`
 - [ ] Operating system: `<REPLACE with e.g., Centos 5 (x64)>`
 - [ ] Provide logs (with `debug=..` as necessary) from librdkafka
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

**Comment by edenhill** [comment]
Please fill out the check-list.

And yes, the typical reason is that the leader is not available.
You can verify this by setting the `debug` config property to `topic,msg`

**Comment by yhsiung** [comment]
git tag: SRCREV = "b677b451700a3548e37298fa070016b5126b71c8"

Is it true the tcping will still work even though the leader is not available?

**Comment by edenhill** [comment]
Yes, the broker process may be up and running, listening to its tcp ports, but the state of the partition can still be that it has no leader.

**Comment by yhsiung** [comment]
I  tried to add the following code to catch the logs from client, but nothing happened. Am I missing anything from the code?

static void log_cb(rd_kafka_t *rk,
        int level, const char *fac, const char *buf)
{
	printf("level: %d, fac: %s, buf: %s", level, fac, buf);
}

main()
{

        /* assume conf and rk are initialized */

        if (rd_kafka_conf_set(conf, "debug", "all", errstr, sizeof(errstr)) != RD_KAFKA_CONF_OK)
        {
                printf("Sets 'debug' failed with %s\n", errstr);
        }

        rd_kafka_conf_set_log_cb(conf, log_cb);
        rd_kafka_set_log_level(rk, 7);

        ....
}

**Comment by edenhill** [comment]
add a newline to your printf, the output may be buffered.
